uswds/uswds
# USWDS - In-Page Navigation: Add data-mininum-heading-count attribute to in-page navigation

# Summary
**Hide in-page navigation when there are fewer than 2 headings .**. The data-minimum-heading-count setting determines when the in-page navigation should be displayed or hidden if the minimum number of headings is not reached.

Previously, in-page navigation was shown even when there were no headings, resulting in an empty component.

## Breaking change
This is not a breaking change.

## Issue
Closes [#5967](https://github.com/uswds/uswds/issues/5967)

## Related Pull Requests
USWDS-Site: Add changelog for in-page navigation [#3000](https://github.com/uswds/uswds-site/pull/3000)

## Problem statement
In scenarios where there is limited content on the page, the in-page navigation does not provide much value.  Currently the in-page navigation is available to the user even if only 1 header was present. This issue can be solved by adding a data attribute to only show the in-page navigation element when the defined minimum number of headers is reached.

## Solution
Provide an attribute called data-minimum-heading-count where the number value sets the minimum amount of headings needed on the page to show the in-page navigation.

## Preview link
https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/ml-inpage-nav-heading-count/?path=/story/components-in-page-navigation--default

## Major changes
-Customizable data attribute `data-minimum-heading-count` added to in-page navigation element
-New test pattern added to display usability
-JS functionality to not render any children in the navigational element when the minimum amount of headers is not met (default or custom setting)

## Related PR

## Testing and review
1. Navigate to the in-page-navigation component
2. Change storybook control for `data-minimum-heading-count` to be higher than the number of headings available
3. Confirm the navigational element is hidden with no children if the minimum heading count has not been reached
4. Change storybook control for `data-minimum-heading-count` to be lower than the number of headings available
5. Confirm it is visible with the correct children if the minimum heading count has been reached
6. Check all controls and visuals for regressions
7. Ensure all tests pass
